thorlcr/activities/wuidwrite/thwuidwriteslave.cpp

```diff
@@ -174,27 +174,24 @@ class CWorkUnitWriteLocalSlaveActivity : public CWorkUnitWriteSlaveBase
                 ActPrintLog("WORKUNITWRITE: exception");
                 throw;
             }
-            if (numGot || totalNum)
```

Comment:
This is the main fix, essentially removing this line, the other changes are just a indentation change
With this condition, it was not communicating with the master if there were 0 total rows, meaning the result was never published.

Reply:
this is not really the main fix actually - it should never have been using the CQ form of workunit write in the context this problem was hit, it was spuriously using this version because of  'grouped=1' in the write activity, which was wrongly interpreted as meaning in CQ context. That is fixed by the other changes.